bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/handler/JablotronJa100Handler.java

```diff
@@ -0,0 +1,220 @@
+        String segmentId = segment.getSegmentId();
+
+        if (segmentId.startsWith("STATE_") || segmentId.startsWith("PGM_")) {
+            processSection(segment);
+        } else if (segmentId.startsWith("THERMOMETER_")) {
+            processThermometer(segment);
+        } else if (segmentId.startsWith("THERMOSTAT_")) {
+            processThermostat(segment);
+        } else {
+            logger.debug("Unknown segment received: {} with state: {}", segment.getSegmentId(),
+                    segment.getSegmentState());
+        }
+    }
+
+    private void processSection(JablotronServiceDetailSegment segment) {
+        String segmentId = segment.getSegmentId().toLowerCase();
+        Channel channel = getThing().getChannel(segmentId);
+        if (channel == null) {
+            logger.debug("Creating a new channel: {}", segmentId);
+            createChannel(segment);
+        }
+        channel = getThing().getChannel(segmentId);
+        if (channel != null) {
+            logger.debug("Updating channel: {} to value: {}", channel.getUID(), segment.getSegmentState());
+            State newState;
+            if (segmentId.startsWith("PGM_")) {
+                newState = "unset".equals(segment.getSegmentState()) ? OnOffType.OFF : OnOffType.ON;
+            } else {
+                newState = new StringType(segment.getSegmentState());
+            }
+            updateState(channel.getUID(), newState);
+        } else {
+            logger.debug("The channel: {} still doesn't exist!", segmentId);
+        }
+    }
+
+    private void processThermometer(JablotronServiceDetailSegment segment) {
+        String segmentId = segment.getSegmentId().toLowerCase();
+        Channel channel = getThing().getChannel(segmentId);
+        if (channel == null) {
+            logger.debug("Creating a new temperature channel: {}", segmentId);
+            createTempChannel(segmentId, segment.getSegmentName());
+        }
+        updateTemperatureChannel(channel, segment);
+    }
+
+    private void processThermostat(JablotronServiceDetailSegment segment) {
+        String segmentId = segment.getSegmentId().toLowerCase();
+        Channel channel = getThing().getChannel(segmentId);
+        if (channel == null) {
+            logger.debug("Creating a new thermostat channel: {}", segmentId);
+            createThermostatChannel(segmentId, segment.getSegmentName());
+        }
+        updateTemperatureChannel(channel, segment);
+    }
+
+    private void updateTemperatureChannel(Channel channel, JablotronServiceDetailSegment segment) {
+        List<JablotronServiceDetailSegmentInfo> infos = segment.getSegmentInfos();
+        if (infos.size() > 0) {
+            logger.debug("Found value: {} and type: {}", infos.get(0).getValue(), infos.get(0).getType());
+            DecimalType newState = new DecimalType(infos.get(0).getValue());
+            updateState(channel.getUID(), newState);
```

Comment:
This needs to be of type `QuantityType`. Please pay attention to use `import org.eclipse.smarthome.core.library.unit.SIUnits`, as there is another class with the same name in a different package.
```suggestion
            updateState(channel.getUID(), QuantityType.valueOf(infos.get(0).getValue(), SIUnits.CELSIUS));
```

Reply:
OK